pkg/arvo/sys/arvo.hoon

```diff
@@ -417,29 +417,67 @@
 ++  de
   =|  fat=(axal)
   |@
+  ++  del
+    |=  pax=path
+    ?~  pax  [~ dir.fat]
+    =/  kid  (~(get by dir.fat) i.pax)
+    ?~  kid  fat
+    $(fat u.kid, pax t.pax)
+  ::  Descend to the axal at this path
   ::
-  ++  get
+  ++  dip
     |=  pax=path
     ^+  fat
     ?~  pax  fat
     =/  kid  (~(get by dir.fat) i.pax)
     ?~  kid  [~ ~]
     $(fat u.kid, pax t.pax)
   ::
-  ++  put
-    |*  [pax=path dat=*]
-    =>  .(dat `_?>(?=(^ fil.fat) u.fil.fat)`dat)
-    ^+  fat
-    ?~  pax  fat(fil `dat)
-    =/  kid  (~(get by dir.fat) i.pax)
-    =/  new  (fall kid fat(fil ~, dir ~))
-    fat(dir (~(put by dir.fat) i.pax $(fat new, pax t.pax)))
-  ::
   ++  gas
     |=  lit=(list (pair path _?>(?=(^ fil.fat) u.fil.fat)))
     ^+  fat
     ?~  lit  fat
     $(fat (put p.i.lit q.i.lit), lit t.lit)
+  ::  Fetch file at longest existing prefix of the path
+  ::
+  ++  get
+    |=  pax=path
+    ^+  [path fat]
+    ?~  pax  [~ fil.fat]
+    =/  kid  (~(get by dir.fat) i.pax)
+    ?~  kid  [pax fil.fat]
+    =/  low  $(fat u.kid, pax t.pax)
+    ?~  +.low
+      [pax fil.fat]
+    low
+  ::
+  ++  has
+    |=  pax=path
+    !=(~ fil:(dip pax))
+  ::  Delete subtree
+  ::
+  ++  lop
+    |=  pax=path
+    ^+  fat
+    ?~  pax  fat
+    |-
+    ?~  t.pax  fat(dir (~(del by dir.fat) i.pax))
+    =/  kid  (~(get by dir.fat) i.pax)
+    ?~  kid  fat
+    fat(dir (~(put by dir.fat) i.pax $(fat u.kid, pax t.pax)))
+  ::  Serialize subppath to map
```

Comment:
subppath -> subpath

Reply:
I don't really understand the description, though.  Isn't this more like "convert trie to map from path to file"?  There's no subpath specified.